src/liberty/libertyProject.ts

```diff
@@ -203,6 +203,49 @@ export class ProjectProvider implements vscode.TreeDataProvider<LibertyProject>
 		statusMessage.dispose();
 	}
 
+	/**
+	 * Method asks the user to save the workspace first if it is untitled and the workspace contains morethan
+	 * one project. if not saved there are chances for the projects state not being saved and manually added
+	 * projects might not be visible in the liberty dashboard.
+	 */
+	public async checkUntitledWorkspaceAndSaveIt(): Promise<void> {
+		return new Promise((resolve) => {
+			try {
+				vscode.window.showInformationMessage(
+					'Please save the workspace first, as manually added projects to the dashboard may not persist in the next VS Code session if the workspace is not saved.',
+					{ modal: true },
+					'Save Workspace'
+				).then(async (selection) => {
+					if (selection === 'Save Workspace') {
+						/**
+						 * setting workspaceSaveInProgress to true and storing it in globalstate for identifyting that the
+						 * workspace is saved and needs to save the manually added projects to the dashboard
+						 */
+						await this._context.globalState.update('workspaceSaveInProgress', true);
+						//opens the saveWorkspace as dialog box
+						await vscode.commands.executeCommand('workbench.action.saveWorkspaceAs');
+					}
+					util.clearDataSavedInGlobalState(this._context);
+					resolve();
+				});
+			} catch (error) {
+				console.debug("exception while saving the workspace" + error);
```

Comment:
if there is an error during saving, should we clear the global state data at this point too?

Reply:
Yes , cleared the global state